### PR TITLE
cli: clean up the code around the handling of default parameters

### PR DESCRIPTION
**Pull request by knz**

### CODE REVIEW

**Comment by knz** [comment]
We might want to cherry pick some (if not all) of this in 1.1. The faulty re-initialization of defaults in between tests has bitten us in the past.

**Comment by knz** [comment]
@tschottdorf @mberhault perhaps one of you could have a look since alex is out?

**Comment by mberhault** [comment]
I'll take a look. At first glance though I would say this doesn't go into 1.1 unless this fixes bugs in non-testing code.

File: pkg/cli/flags.go [hunk 1]
```diff
@@ -20,36 +20,45 @@ import (
...
 	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/log/logflags"
 )
 
-var maxResults int64
-
-var sqlConnURL, sqlConnUser, sqlConnDBName string
-var serverConnHost, serverConnPort, serverAdvertiseHost, serverAdvertisePort string
+// special global variables used by flag variable definitions belows.
```
Comment (mberhault) [inline pkg/cli/flags.go]: s/belows/below/

File: pkg/cli/flags_util.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+	"github.com/cockroachdb/cockroach/pkg/storage/engine"
+	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
+	humanize "github.com/dustin/go-humanize"
+	"github.com/elastic/gosigar"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"golang.org/x/net/context"
+)
+
+// This file contains definitions for data types suitable for use by
```
Comment (mberhault) [inline pkg/cli/flags_util.go]: I don't think this whole section is necessary. Not only is `flag.Value` a subset of the `pflag.Value` interface, types that don't implement `pflag.Value` will spit out a clear error when using them on `VarFlag`.

**Review by mberhault — APPROVED** [review]
LGTM. Thanks for the refactor. Only small nits (that I don't feel strongly about).

No cherry-pick into 1.1 though. If we have specific flag issues in 1.1 we can fix them individually, but definitely not the whole refactor.

**Comment by knz** [comment]
Thanks!

---

Review status: 0 of 8 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/cli/flags.go, line 33 at r11](https://reviewable.io:443/reviews/cockroachdb/cockroach/20987#-L0tg79KDifpislJ53at-r11-33:-L1NdJPyWCKF6Ix0v-CP:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d101af6d3c1f0f1afe7584fb1629839cd23bad7/pkg/cli/flags.go#L33)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

s/belows/below/
</blockquote></details>

Done.

---

*[pkg/cli/flags.go, line 50 at r11](https://reviewable.io:443/reviews/cockroachdb/cockroach/20987#-L0tg79KDifpislJ53at-r11-50:-L1NdJAUbTErpwraRwf-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d101af6d3c1f0f1afe7584fb1629839cd23bad7/pkg/cli/flags.go#L50)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

s/initialize/initializes/
</blockquote></details>

Done.

---

*[pkg/cli/flags_util.go, line 36 at r11](https://reviewable.io:443/reviews/cockroachdb/cockroach/20987#-L0tg79JKBpTzqe0SqS_-r11-36:-L1NdLQjMdufxqZaDBj0:bhgwo7z) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d101af6d3c1f0f1afe7584fb1629839cd23bad7/pkg/cli/flags_util.go#L36)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

I don't think this whole section is necessary. Not only is `flag.Value` a subset of the `pflag.Value` interface, types that don't implement `pflag.Value` will spit out a clear error when using them on `VarFlag`.
</blockquote></details>

Agreed. Removed. (The code was there in different places previously. This simplifies.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20987#-:-L1Nexo4WNo1iNKpsoxV:b-ucfknh)*

**Comment by knz** [comment]
TFYR :)

**Pull request merged**